Source/IGListCollectionViewLayout.mm

```diff
@@ -577,19 +590,23 @@ - (void)calculateLayoutIfNeeded {
                 rollingSectionBounds = CGRectUnion(rollingSectionBounds, frame);
             }
         }
-
+       
         const CGRect headerBounds = (self.scrollDirection == UICollectionViewScrollDirectionVertical) ?
                 CGRectMake(insets.left,
-                        CGRectGetMinY(rollingSectionBounds) - headerSize.height,
+                        (itemCount == 0) ? CGRectGetMaxY(rollingSectionBounds) : CGRectGetMinY(rollingSectionBounds) - headerSize.height,
                         paddedLengthInFixedDirection,
                         headerSize.height) :
-                CGRectMake(CGRectGetMinX(rollingSectionBounds) - headerSize.width,
+                CGRectMake((itemCount == 0) ? CGRectGetMaxX(rollingSectionBounds) : CGRectGetMinX(rollingSectionBounds) - headerSize.width,
                         insets.top,
                         headerSize.width,
                         paddedLengthInFixedDirection);
 
         _sectionData[section].headerBounds = headerBounds;
-
+        
+        if (itemCount == 0) {
```

Comment:
Could we capture `itemCount == 0` in a variable?
```objc
const BOOL itemsEmpty = itemCount == 0;
```

Reply:
Sure. 👍